aptos-move/framework/aptos-stdlib/sources/table.move

```diff
@@ -96,6 +106,41 @@ module aptos_std::table {
     // ======================================================================================================
     // Internal API
 
+    struct NewTableEvent has drop, store {
+        handle: u128,
+        key_type: String,
+        value_type: String,
+    }
+
+    struct TableMetadata has key, store {
+        new_table_events: EventHandle<NewTableEvent>,
+    }
+
+    public fun initialize(account: &signer) {
+        assert!(address_of(account) == @aptos_std, error::permission_denied(ENOT_STD_ACCOUNT));
+
+        assert!(!exists<TableMetadata>(@aptos_std), error::already_exists(ETABLE_METADATA_ALREADY_EXISTS));
+        move_to(account, TableMetadata {
+            new_table_events: new_event_handle<NewTableEvent>(account)
+        });
+    }
+
+    /// Emits NewTableEvent if the event handle is installed under @aptos_std.
+    fun maybe_emit_create_table_event<K: copy + drop, V>(handle: u128) acquires TableMetadata {
```

Comment:
why maybe_? Is it expected to not emit the event outside of unit tests?

Reply:
no, this is just for getting around unittests.

Reply:
Can you make a test_only function to set things up for unit tests? It'd be better if we don't have special logic in code that's deployed to mainnet that's just for making a test work.

Reply:
to be more explicit...

put something in this function

```
#[test_only]
return;
```

of course make sure that works...